### ISSUE TITLE
[Bug]: Invalid site domain validation for domains with umlauts (ä, ö, ü, etc.)

### ISSUE THREAD

**Issue by R-O-BIN** [pimcore/pimcore#17724]
### Pimcore version

11.4.1

### Steps to reproduce

1. Go to site configuration to additional domains
2. Enter a domain that contains umlaut characters such as "ä", "ö", or "ü" (e.g., müller.de).

### Actual Behavior

Domains with umlaut characters are incorrectly validated.

### Expected Behavior

Domains containing umlaut characters  should be treated as valid in accordance with international domain name (IDN) standards.

**Comment by kingjia90** [comment]
Thank you for reporting!

`FILTER_VALIDATE_URL`  could be not ideal due to https://bugs.php.net/bug.php?id=42353
@blankse any tips?

**Comment by blankse** [comment]
@kingjia90 I think we could encode the umlauts:
```
public function validateUrl(string $host): bool
{
    $sanitizedHost = '';

    foreach (str_split($host) as $character) {
        $sanitizedHost .= (ord($character) > 127) ? rawurlencode($character) : $character;
    }

    return filter_var($sanitizedHost, FILTER_VALIDATE_DOMAIN, FILTER_FLAG_HOSTNAME) !== false;
}
```

**Comment by jdreesen** [comment]
There's also [`idn_to_ascii`](https://www.php.net/manual/en/function.idn-to-ascii.php) in `intl`:

```php
$uri = 'http://müller.de';

$host = idn_to_ascii(parse_url($uri, PHP_URL_HOST), IDNA_NONTRANSITIONAL_TO_ASCII);
// or a little more strict:
$host = idn_to_ascii(parse_url($uri, PHP_URL_HOST), IDNA_NONTRANSITIONAL_TO_ASCII|IDNA_CHECK_BIDI|IDNA_USE_STD3_RULES|IDNA_CHECK_CONTEXTJ);

// host = xn--mller-kva.de

var_dump(filter_var($host, FILTER_VALIDATE_DOMAIN, FILTER_FLAG_HOSTNAME) !== false);
// => true
```

**Comment by kingjia90** [comment]
Thank you both, the approach suggested by @jdreesen seems what we need here and might cover more cases that we may have missed (eg. like chinese/cyrillic characters?)

Could you please open a PR for it?

**Comment by blankse** [comment]
@kingjia90 I created a PR: https://github.com/pimcore/pimcore/pull/17734

**Comment by jdreesen** [comment]
So can this be closed now that #17734 has been merged?

**Comment by kingjia90** [comment]
Fixed by #17734
